wordpress-mobile/WordPress-Android
# Dashboard Cards: Activity Card implement onclick behavior

Closes #18217 

This PR implements the click actions for the ActivityLog dashboard card
- On footer row click (View All Activity), navigate to Activity Log view
- On activity card item click, navigate to Activity detail view
- Modifications were made to the activity item onClick handler to allow for a multiple parameters in the form of `ActivityCardItemClickParams` instead of a single string. 
- `ActivityLogDetailViewModel` was refactored to check for the presence of activityId in activity log table first and then check the dashboard cards table (if entry was from the activity card)
- Multiple refactors to `ActivityLogDetailViewModelTest` to support the changes in `ActivityLogDetailViewModel

Notes:
- A decision was made that when accessing the detail view via the activity card, the rewind/backup-download buttons will be managed by the isRewinable attribute on the ActivityLog item itself. If there's a restore / backup running already, the user will still be able to see and tap the restore / backup buttons, but they'll get an error message when they attempt to kick it off. This is preexisting behavior.
- Event tracking will follow in a separate PR


**To test:**
**Activity Detail View is shown when tapping on activity card item BEFORE visiting activity log list**
1. Install the Jetpack app (do a fresh install - want to make sure the DB is clear)
2. Login with a wp account
3. Select a site in which you can access the activity log (manage content/admin)
4. Navigate to App Setting > Debug Settings 
5. Enable `dashboard_card_activity_log` and restart the app
6. Navigate to Home tab
7. Scroll to the activity card 
8. Tap on an activity item 
9. ✅ Verify you are taken to the Activity Detail view
----------------
**Activity Log list is shown when tapping on Footer link**
1. Continue from test above
2. Navigate back to the home tab
4. Scroll to the activity card 
5. Tap on an footer link "View all activity"
6. ✅ Verify you are taken to the Activity Log list view
---------------
**Activity Detail View is shown when tapping on activity card item AFTER visiting activity log list**
1. Continue from test above
2. Navigate back to the home tab
3. Scroll to the activity card 
4. Tap on an activity item 
5. ✅ Verify you are taken to the Activity Detail view

## Regression Notes
1. Potential unintended areas of impact
Activity Log detail view doesn't function correctly when accessed via the activity dashboard card

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and new unit test

4. What automated tests I added (or what prevented me from doing so)
Add tests to `ActivityLogDetailViewModelTest`

PR submission checklist:
